### PR TITLE
Redesign task switcher

### PR DESCRIPTION
**Pull request by markfijneman**

### CODE REVIEW

File: front/src/components/Task/TaskTagChip.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    let clickable;
+
+    // Check if keys.filterTag exists
+    try {
+      filterTag = injectStrict(keys.filterTag);
+      clickable = true;
+    } catch (e) {
+      filterTag = () => {return null;};
+      clickable = false;
+    }
```
Comment (JJ-8) [inline front/src/components/Task/TaskTagChip.vue:38]: What is the case that it does not exist?

File: front/src/components/Task/TaskTagsListCondensed.vue [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    </q-chip>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { Tag } from 'src/ctfnote/models';
+import TaskTagChip from './TaskTagChip.vue';
+
+export default defineComponent({
```
Comment (JJ-8) [inline front/src/components/Task/TaskTagsListCondensed.vue:50]: Just for my understanding: the TaskTagsListCondensed is the version with the '+1' and the TaskTagsList is the original version and you have chosen to not merge those two implementations into one file with a boolean switch to enable/disable the '+1'? Is there any reason to not do that?

**Review by JJ-8 — CHANGES_REQUESTED** [review]
This looks so fancy!
image

Really like the '+X' popup! It also handles lots of tags perfectly fine. However, the black color causes a low contrast in the dark mode. Light mode has sufficient contrast of course. Maybe use a different color as a default color for this (just a normal color)?

File: front/src/components/Task/TaskTagChip.vue (hunk 1 above)
Comment (markfijneman) [reply front/src/components/Task/TaskTagChip.vue:38]: This only exists in the main task list. This code (a large part of which already existed prior to this pull request) adds clicked tags to 'Filter by tag', and that filter does not exist in the task switcher dropdown. Without this statement Vue can't find `keys.filterTag` and would refuse to render the dropdown menu.

File: front/src/components/Task/TaskTagsListCondensed.vue (hunk 2 above)
Comment (markfijneman) [reply front/src/components/Task/TaskTagsListCondensed.vue]: Good point. They are now merged into a single class.

**Review by JJ-8 — APPROVED** [review]

**Pull request merged**